### PR TITLE
Adds pre-commit checks to the repo -- SGPROJ-1573

### PR DESCRIPTION
**Pull request by DJKaswa**
## Change description

> Please review this PR. It passes all checks locally. Feel free to fix anything that will block it from passing on Github, otherwise it should be good to go.

> Once you review and approve it, you can merge it to main too (don’t forget to delete the branch). At that point, it’ll be ready to be moved to tf-auto

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

File: lambda.tf [hunk 1]
```diff
@@ -6,6 +6,7 @@ resource "aws_lambda_permission" "apigw_lambda" {
   source_arn    = "arn:aws:execute-api:${var.region}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.sftp.id}/*/${aws_api_gateway_method.get.http_method}${aws_api_gateway_resource.config.path}"
 }
 
+#tfsec:ignore:aws-lambda-enable-tracing -- Ignores warning on Function not having tracing enabled
```
Comment (JoseMotto-sg) [inline lambda.tf:9]: Please enable logs for the resource
```suggestion
```

File: main.tf [hunk 2]
```diff
@@ -27,6 +23,7 @@ EOF
   )
 }
 
+#tfsec:ignore:aws-iam-no-policy-wildcards -- Ignores warning on Overly permissive policies
```
Comment (JoseMotto-sg) [inline main.tf:26]: Please setup correct permissions on policy
```suggestion
```

**Comment by DJKaswa** [comment]
@JoseMotto-sg please review again

**Review by JoseMotto-sg — APPROVED** [review]
LGTM

**Pull request merged**